### DISCUSSION TITLE
AlertManager Datasources should also be selectable as external AlertManagers

### DISCUSSION THREAD

**Discussion by siegenthalerroger** [grafana/grafana#42660] [Alerting]
Currently it is possible to add external AlertManagers by URL add have all alerts forwarded to an (or multiple) AMs. It will also be possible to completely deactivate the built-in AM (https://github.com/grafana/grafana/issues/40295).
This is great.

However, there is also another way to add external AlertManagers to Grafana, and that is by adding an AM datasource. This provides different capabilities (visualising configuration of AM and dynamic configuration if using cortex) but lacks the above capability to have all alerts forwarded to it. This request addresses this discrepancy.

**Allow AlertManager datasources to be set as an external AM.** Thus allowing one configuration of an AM to serve both a visualisation/config and forwarding role. There's also the security aspect of passwords being raw-text in the current external AM configuration vs secureStorage in the datasource option.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
